conans/test/unittests/tools/gnu/autotoolschain_test.py

```diff
@@ -211,3 +212,18 @@ def test_update_or_prune_any_args(cross_building_conanfile):
     at.update_make_args({"--new-complex-flag": "new-value"})
     new_make_args = cmd_args_to_string(at.make_args)
     assert "--new-complex-flag=new-value" in new_make_args
+
+
+def test_pkg_config_path():
```

Comment:
This test is passing in Windows, without your changes above.
We would need first a failing test that illustrate the bugfix, then fix it.

Reply:
Yes, I opened another PR only adding this test and was green, I was expecting it to be red. It seems I need to run tests locally first.

I am sure that this PR is making ffmpeg and similar packages running again. I need more time to setup the test environment on Windows and Msys2.

Reply:
Is the above test running on Windows with MSCVC?

Reply:
Yes, green on windows (msvc is irrelevant, it is a unittest, and the settings are mocked to be msvc, it will be the same in every platform)

Reply:
Probably you are right that the particular toolchain is not relevant, I wanted anyway to focus on the combination of `MSVC`, `AutoToolsDeps`, `AutotoolsToolchain` and `Msys2`. This is the particular configuration common in many packages like ffmpeg (and other packages) that have to implement [this](https://github.com/conan-io/conan-center-index/blob/master/recipes/ffmpeg/all/conanfile.py#L656) "hack" for AutoToolsDeps. 

After testing this specific scenario, I can try something more generic just like `AutoToolchain` and `Msys2` on Windows. I am not aware of any issue on Linux and Darwin systems.

Reply:
I think I am a bit closer to understand the issue. When calling `generate` method on `AutoToolsToolchain` I see that the `conanautotoolstoolchain.sh` looks like:

```Shell
script_folder="C:\Users\edukaj\projects\dev\repos\conan\conans\test\unittests\tools\gnu\generators"
echo "echo Restoring environment" > "$script_folder\deactivate_conanautotoolstoolchain.sh"
for v in CPPFLAGS CXXFLAGS CFLAGS LDFLAGS PKG_CONFIG_PATH
do
    is_defined="true"
    value=$(printenv $v) || is_defined="" || true
    if [ -n "$value" ] || [ -n "$is_defined" ]
    then
        echo export "$v='$value'" >> "$script_folder\deactivate_conanautotoolstoolchain.sh"
    else
        echo unset $v >> "$script_folder\deactivate_conanautotoolstoolchain.sh"
    fi
done


export CPPFLAGS="$CPPFLAGS -DNDEBUG"
export CXXFLAGS="$CXXFLAGS -MD -O2 -Ob2 -FS"
export CFLAGS="$CFLAGS -MD -O2 -Ob2 -FS"
export LDFLAGS="$LDFLAGS"
export PKG_CONFIG_PATH="/c/users/edukaj/projects/dev/repos/conan/conans/test/unittests/tools/gnu/generators:$PKG_CONFIG_PATH"
```

Probably on MSYS2, when `PKG_CONFIG_PATH` is not defined this is expanded to `/some/path:` leading to the issue I am seeing.

Reply:
> Probably on MSYS2, when PKG_CONFIG_PATH is not defined this is expanded to /some/path: leading to the issue I am seeing.

But a trailing ``:`` (or several), separator doesn't break PATHs in linux path variables, not sure why this could be an issue, but please let us know if you figure out!

Reply:
I can confirm that on Linux it works while on Msys2 it breaks. 

```
$ export PKG_CONFIG_PATH=$PWD:
/c/Users/edukaj/.conan2/p/b/ffmpe0234e7cab6ab1/b/build-release/conan:

$ pkg-config --libs 'aom >= 1.0.0'
Package aom was not found in the pkg-config search path.
Perhaps you should add the directory containing `aom.pc'
to the PKG_CONFIG_PATH environment variable
Package 'aom', required by 'virtual:world', not found

$ export PKG_CONFIG_PATH=$PWD

$ echo $PKG_CONFIG_PATH
/c/Users/edukaj/.conan2/p/b/ffmpe0234e7cab6ab1/b/build-release/conan

$ pkg-config --libs 'aom >= 1.0.0'
-LC:/Users/edukaj/.conan2/p/b/libao25de1f6d5f9ce/p/lib -laom
```

This is something very easy to test on msys2. But at this point seems more a bug on msys2 rather than on `conan`.

Reply:
Maybe it's an issue of `pkgconf` package on Windows. I will give a look to that package first. I also repeated the tests using the same pkgconf and msys2 installed by conan.

And I can still replicate the bug.

```
PS C:\Users\edukaj\.conan2\p\b\msys27d3b492158843\p\bin\msys64\usr\bin> .\bash.exe -l -c 'echo pkgconf version: $(/c/Users/edukaj/.conan2/p/pkgcoe6fcefd89afdc/p/bin/pkgconf --version); echo "TEST WITH TRAILING :"; PKG_CONFIG_PATH=/c/Users/edukaj/.conan2/p/b/ffmpe0234e7cab6ab1/b/build-release/conan: ; export PKG_CONFIG_PATH; echo PKG_CONFIG_PATH is $PKG_CONFIG_PATH; /c/Users/edukaj/.conan2/p/pkgcoe6fcefd89afdc/p/bin/pkgconf --libs aom; echo "TEST WITH NO TRAILING :"; PKG_CONFIG_PATH=/c/Users/edukaj/.conan2/p/b/ffmpe0234e7cab6ab1/b/build-release/conan; export PKG_CONFIG_PATH; echo PKG_CONFIG_PATH is $PKG_CONFIG_PATH; /c/Users/edukaj/.conan2/p/pkgcoe6fcefd89afdc/p/bin/pkgconf --libs aom; '
mkdir: cannot create directory '/dev/shm': Read-only file system

Creating /dev/shm directory failed.
POSIX semaphores and POSIX shared memory will not work

mkdir: cannot create directory '/dev/mqueue': Read-only file system

Creating /dev/mqueue directory failed.
POSIX message queues will not work

pkgconf version: 2.1.0
TEST WITH TRAILING :
PKG_CONFIG_PATH is /c/Users/edukaj/.conan2/p/b/ffmpe0234e7cab6ab1/b/build-release/conan:
Package aom was not found in the pkg-config search path.
Perhaps you should add the directory containing `aom.pc'
to the PKG_CONFIG_PATH environment variable
Package 'aom', required by 'virtual:world', not found
TEST WITH NO TRAILING :
PKG_CONFIG_PATH is /c/Users/edukaj/.conan2/p/b/ffmpe0234e7cab6ab1/b/build-release/conan
-LC:/Users/edukaj/.conan2/p/b/libao25de1f6d5f9ce/p/lib -laom
```

Reply:
Other tests:



```
ls C:\users\edukaj\.conan2\p\b\ffmpeb0f324ccbd53c\b\build-release\conan\

----                 -------------         ------ ----
-a---           4/15/2024  6:11 PM            237 aom.pc
-a---           4/15/2024  6:11 PM            287 brotli.pc
-a---           4/15/2024  6:11 PM            239 bzip2.pc
-a---           4/15/2024  6:11 PM           4178 conanautotoolsdeps_cl_workaround.sh
-a---           4/15/2024  6:11 PM            819 conanautotoolstoolchain.sh
-a---           4/15/2024  6:11 PM             39 conanbuild.bat
-a---           4/15/2024  6:11 PM           1299 conanbuild.conf
-a---           4/15/2024  6:11 PM            425 conanbuild.sh
-a---           4/15/2024  6:11 PM           1307 conanbuildenv-release-x86_64.sh
-a---           4/15/2024  6:11 PM            626 conanrunenv-release-x86_64.sh
-a---           4/15/2024  6:11 PM            330 conanvcvars.bat
-a---           4/15/2024  6:11 PM            241 dav1d.pc
-a---           4/15/2024  6:13 PM            156 deactivate_conanautotoolsdeps_cl_workaround.sh
-a---           4/15/2024  6:13 PM            158 deactivate_conanautotoolstoolchain.sh
-a---           4/15/2024  6:11 PM             50 deactivate_conanbuild.bat
-a---           4/15/2024  6:11 PM            469 deactivate_conanbuild.sh
-a---           4/15/2024  6:13 PM           2726 deactivate_conanbuildenv-release-x86_64.sh
-a---           4/15/2024  6:13 PM             49 deactivate_conanrunenv-release-x86_64.sh
-a---           4/15/2024  6:11 PM             67 deactivate_conanvcvars.bat
-a---           4/15/2024  6:12 PM             98 deactivate_msys2_mode.bat
-a---           4/15/2024  6:11 PM            265 fdk-aac-fdk-aac.pc
-a---           4/15/2024  6:11 PM            263 fdk-aac.pc
-a---           4/15/2024  6:11 PM            349 freetype2.pc
-a---           4/15/2024  6:11 PM            333 libbrotlicommon.pc
-a---           4/15/2024  6:11 PM            350 libbrotlidec.pc
-a---           4/15/2024  6:11 PM            350 libbrotlienc.pc
-a---           4/15/2024  6:11 PM            278 libcpuinfo-cpuinfo.pc
-a---           4/15/2024  6:11 PM            279 libcpuinfo.pc
-a---           4/15/2024  6:11 PM            328 libcrypto.pc
-a---           4/15/2024  6:11 PM            256 libiconv.pc
-a---           4/15/2024  6:11 PM            262 liblzma.pc
-a---           4/15/2024  6:11 PM            253 libmp3lame.pc
-a---           4/15/2024  6:11 PM            328 libopenjp2.pc
-a---           4/15/2024  6:11 PM            271 libpng.pc
-a---           4/15/2024  6:11 PM            144 libpng16.pc
-a---           4/15/2024  6:11 PM            281 libsharpyuv.pc
-a---           4/15/2024  6:11 PM            276 libssl.pc
-a---           4/15/2024  6:11 PM            271 libsvtav1.pc
-a---           4/15/2024  6:11 PM            336 libwebp-all-do-not-use.pc
-a---           4/15/2024  6:11 PM            291 libwebp.pc
-a---           4/15/2024  6:11 PM            290 libwebpdecoder.pc
-a---           4/15/2024  6:11 PM            302 libwebpdemux.pc
-a---           4/15/2024  6:11 PM            296 libwebpmux.pc
-a---           4/15/2024  6:11 PM            245 libx264.pc
-a---           4/15/2024  6:12 PM            637 msys2_mode.bat
-a---           4/15/2024  6:11 PM            239 ogg.pc
-a---           4/15/2024  6:11 PM            250 openh264.pc
-a---           4/15/2024  6:11 PM            266 openssl.pc
-a---           4/15/2024  6:11 PM            296 opus.pc
-a---           4/15/2024  6:11 PM            302 SvtAv1Dec.pc
-a---           4/15/2024  6:11 PM            302 SvtAv1Enc.pc
-a---           4/15/2024  6:11 PM            319 vorbis-all-do-not-use-vorbisenc-alias.pc
-a---           4/15/2024  6:11 PM            322 vorbis-all-do-not-use-vorbisfile-alias.pc
-a---           4/15/2024  6:11 PM            380 vorbis-all-do-not-use.pc
-a---           4/15/2024  6:11 PM            282 vorbis.pc
-a---           4/15/2024  6:11 PM            294 vorbisenc.pc
-a---           4/15/2024  6:11 PM            297 vorbisfile.pc
-a---           4/15/2024  6:11 PM            238 vpx.pc
-a---           4/15/2024  6:11 PM            245 x264.pc
-a---           4/15/2024  6:11 PM            236 x265.pc
-a---           4/15/2024  6:11 PM            240 zlib.pc
.\usr\bin\bash.exe -lc "PKG_CONFIG_PATH=/c/users/edukaj/.conan2/p/b/ffmpeb0f324ccbd53c/b/build-release/conan:/mingw64/lib/pkgconfig:/mingw64/share/pkgconfig /c/users/edukaj/.conan2/p/b/pkgcoc3a2a2e9616d9/p/bin/pkgconf.exe --exists --print-errors zlib"

.\usr\bin\bash.exe -lc "PKG_CONFIG_PATH=/c/users/edukaj/.conan2/p/b/ffmpeb0f324ccbd53c/b/build-release/conan:/mingw64/lib/pkgconfig::/mingw64/share/pkgconfig /c/users/edukaj/.conan2/p/b/pkgcoc3a2a2e9616d9/p/bin/pkgconf.exe --exists --print-errors zlib"
Package zlib was not found in the pkg-config search path.
Perhaps you should add the directory containing `zlib.pc'
to the PKG_CONFIG_PATH environment variable
Package 'zlib' not found
```